armi/reactor/assemblies.py

```diff
@@ -199,6 +204,11 @@ def moveTo(self, locator):
 
     def insert(self, index, obj):
         """Insert an object at a given index position with the assembly."""
+        if self.blockType is not None and not isinstance(obj, self.blockType):
+            msg = f"Cannot insert {obj} to this Assembly, it is not a {self.blockType}."
+            runLog.error(msg)
+            raise TypeError(msg)
```

Comment:
Similar non-blocking request for `self._checkPotentialChild`

Reply:
Done